app/controllers/spina/registers_controller.rb

```diff
@@ -34,14 +34,18 @@ def show
       if params[:status]
         case params[:status]
         when 'closed'
-          @records = Kaminari.paginate_array(@register_data.get_expired_records).page(params[:page]).per(100)
+          @records = paginate(@register_data.get_expired_records)
         when 'current'
-          @records = Kaminari.paginate_array(@register_data.get_current_records).page(params[:page]).per(100)
+          @records = paginate(@register_data.get_current_records)
         when 'all'
-          @records = Kaminari.paginate_array(@register_data.get_records).page(params[:page]).per(100)
+          @records = paginate(@register_data.get_records)
         end
       else
-        @records = Kaminari.paginate_array(@register_data.get_current_records).page(params[:page]).per(100)
+        @records = paginate(@register_data.get_current_records)
+      end
+
+      if params[:q]
```

Comment:
As a user, how might I search only current or expired records with a search term?

Reply:
Right now as I understand it this is out of the scope of the requirements. You can either search all records or filter records. You can't combine them.

Reply:
OK I spoke with @elliecraven and connor and they want the filter to apply to the search, so I will pick that up.

Reply:
👍